vars/neoDeploy.groovy

```diff
@@ -8,222 +11,218 @@ import com.sap.piper.tools.ToolDescriptor
+            .withMandatoryProperty('neo', 'Error in neoDeploy: neo not configured.')
+            .withMandatoryProperty('source', 'Error in neoDeploy: source not configured.')
+            .withMandatoryProperty('deployMode', 'Error in neoDeploy: deployMode not configured.')
+            .withMandatoryProperty('warAction', 'Error in neoDeploy: warAction not configured.')
             .use()
 
         utils.pushToSWA([
-            step: STEP_NAME,
-            stepParam1: configuration.deployMode == 'mta'?'mta':'war', // ['mta', 'warParams', 'warPropertiesFile']
-            stepParam2: configuration.warAction == 'rolling-update'?'blue-green':'standard', // ['deploy', 'deploy-mta', 'rolling-update']
+            step      : STEP_NAME,
+            stepParam1: configuration.deployMode == 'mta' ? 'mta' : 'war', // ['mta', 'warParams', 'warPropertiesFile']
+            stepParam2: configuration.warAction == 'rolling-update' ? 'blue-green' : 'standard', // ['deploy', 'deploy-mta', 'rolling-update']
             stepParam3: parameters?.script == null,
-            stepParam4: ! stepCompatibilityConfiguration.isEmpty(),
+            stepParam4: !stepCompatibilityConfiguration.isEmpty(),
         ], configuration)
 
-        def archivePath = configuration.archivePath
-        if(archivePath?.trim()) {
-            if (!fileExists(archivePath)) {
-                error "Archive cannot be found with parameter archivePath: '${archivePath}'."
+        ToolDescriptor neo = new ToolDescriptor('SAP Cloud Platform Console Client', 'NEO_HOME', 'neoHome', '/tools/', 'neo.sh', null, 'version')
+        ToolDescriptor java = new ToolDescriptor('Java', 'JAVA_HOME', '', '/bin/', 'java', '1.8.0', '-version 2>&1')
+
+        if (configuration.neo.credentialsId) {
+            withCredentials([usernamePassword(
+                credentialsId: configuration.neo.credentialsId,
+                passwordVariable: 'NEO_PASSWORD',
+                usernameVariable: 'NEO_USERNAME')]) {
+
+                assertPasswordRules(NEO_PASSWORD)
+                String neoExecutable = neo.getToolExecutable(script, configuration)
+                NeoCommandHelper neoCommandHelper = new NeoCommandHelper(
+                    script,
+                    configuration.deployMode,
+                    configuration.neo,
+                    neoExecutable,
+                    NEO_USERNAME,
+                    NEO_PASSWORD,
+                    configuration.source
+                )
+
+                dockerExecute(
+                    script: script,
+                    dockerImage: configuration.get('dockerImage'),
+                    dockerEnvVars: configuration.get('dockerEnvVars'),
+                    dockerOptions: configuration.get('dockerOptions')
+                ) {
+
+                    neo.verify(this, configuration)
+                    java.verify(this, configuration)
+
+                    lock("$STEP_NAME :${neoCommandHelper.resourceLock()}") {
```

Comment:
IMO would be better to see what goes into the lock rather than retrieving the infos inside the method. I just asked me if there is the same lock returned each time (since the method is parameterless). That would be quite un-usable since we can parallel deployments into different accounts.

Reply:
The resource look depends on the configuration provided to the neoCommandHelper. It usually takes the host and account into account and for war deployments the application name. However, the method of extracting this data is quite "complex" and moved into this method.